scanner/src/test/kotlin/utils/FileListResolverTest.kt

```diff
@@ -22,40 +22,39 @@ package org.ossreviewtoolkit.scanner.utils
 import io.kotest.core.spec.Spec
 import io.kotest.core.spec.style.StringSpec
 import io.kotest.engine.spec.tempdir
+import io.kotest.matchers.file.aDirectory
 import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
 
 import java.io.File
 
 import org.ossreviewtoolkit.model.ArtifactProvenance
 import org.ossreviewtoolkit.model.RemoteArtifact
 import org.ossreviewtoolkit.model.toYaml
 import org.ossreviewtoolkit.model.utils.FileProvenanceFileStorage
+import org.ossreviewtoolkit.utils.ort.createOrtTempDir
 import org.ossreviewtoolkit.utils.ort.storage.LocalFileStorage
 
 class FileListResolverTest : StringSpec({
     "resolve() should create the expected file list" {
+        val dir = createTempDirWithFiles(".git/index", "LICENSE", "src/cli/main.cpp")
         val resolver = FileListResolver(
             storage = FileProvenanceFileStorage(
                 storage = LocalFileStorage(tempdir()),
                 filename = "bytes"
             ),
-            provenanceDownloader = {
-                createTestTempDirWithFiles(
-                    ".git/index",
-                    "LICENSE",
-                    "src/cli/main.cpp"
-                )
-            }
+            provenanceDownloader = { dir }
         )
 
         val fileList = resolver.resolve(ArtifactProvenance(sourceArtifact = RemoteArtifact.EMPTY))
 
         fileList.toYaml() shouldBe File("src/test/assets/expected-file-list.yml").readText()
+        dir shouldNotBe aDirectory()
```

Comment:
Can we have a dedicated test which verifies that the directory is deleted, instead of adding the assertion to the existing unrelated test case ?